### PR TITLE
Issue 578 (keep legend scrollbar position after toggle)

### PR DESCRIPTION
**Pull request by n-riesco**

### CODE REVIEW

**Comment by etpinard** [comment]
In commit 8f053164a643ebf49e1e6887f05cd48c194b3f9b, you mentioned 

> Only register the wheel and drag listeners during first render.

This can't be _just_ on first render. Consider, this case:

```js
// in the test dashboard console

Tabs.plotMock('legend_scroll');

// then
var gd = Tabs.getGraph();
Plotly.relayout(gd, 'showlegend', false);

// then 
Plotly.relayout(gd, 'showlegend', true);
```

leads to an un-scrollable legend.

So, I think you'll need to relax the on-first-render requirement to something like every time the legend group enter selection is non-empty.

**Comment by n-riesco** [comment]
@etpinard OK, I'll fix that and add a test for it.

**Comment by n-riesco** [comment]
PR is ready for review again.

**Comment by etpinard** [comment]
@n-riesco getting there, but looks like the scroll bar does not appear after a `relayout` call. 

gifrecord_2016-06-02_111721


Scrolling works though and makes the scroll bar appear.

**Comment by n-riesco** [comment]
@etpinard Well, finally the penny dropped! And I understood that it makes more sense to use the legend enter selection as you suggested.

File: src/components/legend/draw.js [hunk 1]
```diff
@@ -122,7 +119,8 @@ module.exports = function draw(gd) {
                 .call(setupTraceToggle, gd);
         });
 
-    if(gd.firstRender) {
+    var firstRender = legend.enter().size() !== 0;
```
Comment (etpinard) [inline src/components/legend/draw.js]: nicely done.

**Comment by etpinard** [comment]
> BTW, what did you use to create the gif?

I'm using this [`script`](http://askubuntu.com/a/201018). Not the cleanest solution. But it gets the job done.

**Comment by etpinard** [comment]
Great PR. Thanks @n-riesco

**Comment by john-soklaski** [comment]
This works great. Thanks @n-riesco !

**Pull request merged**